bindings/pyroot/pythonizations/python/ROOT/_pythonization/_tmva/_batchgenerator.py

```diff
@@ -326,15 +334,10 @@ def ConvertBatchToNumpy(self, batch: "RTensor") -> np.ndarray:
                     self.weights_index -= 1
 
                 weights_data = return_data[:, self.weights_index]
-                return_data = np.column_stack(
-                    (
-                        return_data[:, : self.weights_index],
-                        return_data[:, self.weights_index + 1 :],
-                    )
-                )
-                return return_data, target_data, weights_data
+                return_data = np.column_stack((return_data[:, : self.weights_index], return_data[:, self.weights_index + 1 :]))
```

Comment:
This means that the `return_data` is computed twice in case the `if self.weights_given` is true. The logic can be improved